app/controllers/profiles_controller.rb

```diff
@@ -3,7 +3,15 @@ class ProfilesController < ApplicationController
 
   def update
     @user.profile.update(profile_params)
-    redirect_back(fallback_location: edit_user_path(@user))
+    if @user.spam?
+      @user.is_spam = @user.spam?
```

Comment:
The `.spam?` call makes an API request here, so we'd probably want to avoid 2 requests and do
```suggestion
      @user.is_spam = true
```

Reply:
Down the line, we could (should?) maybe make this a background jerb to avoid a network request in our request cycle. But we'd have to think through the logic on the UI side (maybe we show the flash[:error] on every request by the logged in user where `is_spam` is true?)